### PR TITLE
master(dm): add method check worker status

### PR DESCRIPTION
**Pull request by fatelei**
### What problem does this PR solve? 
#3254 

### What is changed and how it works?
for op is stop-task, add method to check worker status

### Check List 

Tests 

 maybe not need a new tests?

### Related changes

- add method to check worker status when op is stop-task

### CODE REVIEW

**Comment by fatelei** [comment]
I can do tests in local pc ?

**Comment by lance6716** [comment]
Hi, `stop-task` wants the task to be "not found (stopped)", there's nothing to do with worker status

File: dm/dm/master/server.go [hunk 1]
```diff
@@ -2421,3 +2423,18 @@ func (s *Server) sharedLogic(ctx context.Context, req interface{}, respPointer i
 	*errPointer = terror.ErrMasterRequestIsNotForwardToLeader
 	return true
 }
+
+// checkWorkerStatus when op is stop-task, need check worker status.
```
Comment (Ehco1996) [inline dm/dm/master/server.go:2427]: we also use this func when `check whether source relative worker has been removed by scheduler` so please adjust this comment

File: dm/dm/master/server.go [hunk 2]
```diff
@@ -1629,14 +1630,9 @@ func (s *Server) waitOperationOk(
...
 			if expect == pb.Stage_Stopped {
-				resp := &pb.QueryStatusResponse{
-					Result:       true,
-					SourceStatus: &pb.SourceStatus{Source: sourceID, Worker: cli.BaseInfo().Name},
-				}
-				if w := s.scheduler.GetWorkerByName(cli.BaseInfo().Name); w == nil {
-					return true, "", resp, nil
-				} else if cli.Stage() == scheduler.WorkerOffline {
-					return true, "", resp, nil
+				workerQueryResponse = s.checkWorkerStatus(sourceID, cli)
```
Comment (Ehco1996) [inline dm/dm/master/server.go:1633]:
```suggestion
				if workerQueryResponse = s.checkWorkerStatus(sourceID, cli); workerQueryResponse!=nil{
				    return xxx
			
				}
```

File: dm/dm/master/server.go [hunk 3]
```diff
@@ -1697,8 +1696,11 @@ func (s *Server) waitOperationOk(
 							finished = pb.Stage_Finished
 						}
 						if expect == pb.Stage_Stopped {
+							workerQueryResponse = s.checkWorkerStatus(sourceID, cli)
 							if st, ok2 := subtaskStatus.Status.(*pb.SubTaskStatus_Msg); ok2 && st.Msg == dmcommon.NoSubTaskMsg(taskName) {
-								ok = true
+								if workerQueryResponse != nil {
```
Comment (Ehco1996) [inline dm/dm/master/server.go:1701]: semms we can mv L1699 here

File: dm/dm/master/server.go [hunk 4]
```diff
@@ -1682,7 +1678,10 @@ func (s *Server) waitOperationOk(
 				}
 			case *pb.StartTaskRequest, *pb.UpdateTaskRequest, *pb.OperateTaskRequest:
 				if expect == pb.Stage_Stopped && len(queryResp.SubTaskStatus) == 0 {
-					return true, "", queryResp, nil
+					workerQueryResponse = s.checkWorkerStatus(sourceID, cli)
```
Comment (Ehco1996) [inline dm/dm/master/server.go:1681]: i am not sure if this fix https://github.com/pingcap/ticdc/issues/3254, could you mock `task has an error` in `TestWaitOperationOkRightResult` to verify this patch works fine?

**Pull request closed without merging**